deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkHandler.java

```diff
@@ -26,7 +26,7 @@
  * incoming {@code Intent} to the correct Activities annotated with {@link DeepLink}.
  */
 @Target({ ElementType.TYPE })
-@Retention(RetentionPolicy.SOURCE)
+@Retention(RetentionPolicy.CLASS)
```

Comment:
seems like we should add a check somewhere that all annotations for `DeepLinkSpec` should also have non source retention

Reply:
Gradle actually does this for you. If you reference any classes that use `SOURCE` retention in `getSupportedAnnotationTypes()`, then Gradle will fail when attempting to run the processor:

```
> Task :sample-library:compileDebugJavaWithJavac FAILED
error: '@DeepLinkModule' has source retention. Aggregating annotation processors require class or runtime retention.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sample-library:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

It might not be worth adding an additional check?

Reply:
sure, sounds like relying on the gradle error would be simpler, thanks for the details!